apps/coscrad-frontend/src/utils/generic-components/presenters/tables/generic-index-table-presenter/index-table.tsx

```diff
@@ -23,18 +38,46 @@ export interface GenericIndexTablePresenterProps<T extends IBaseViewModel> {
     tableData: T[];
     cellRenderersDefinition: CellRenderersDefinition<T>;
     heading: string;
+    filterableProperties?: (keyof T)[];
 }
 
 export const IndexTable = <T extends IBaseViewModel>({
     headingLabels,
     tableData,
     cellRenderersDefinition,
     heading,
+    filterableProperties = [],
 }: GenericIndexTablePresenterProps<T>) => {
     if (headingLabels.length === 0) {
         throw new EmptyIndexTableException();
     }
 
+    const [searchValue, setSearchValue] = useState('');
+
+    // SEARCH LOGIC
+    const filteredTableData =
+        searchValue.length === 0
+            ? tableData
+            : tableData.filter((row) =>
+                  filterableProperties.some((propertyKey) =>
+                      String(row[propertyKey]).toLowerCase().includes(searchValue.toLowerCase())
```

Comment:
Make the search strategy injectable.

Reply:
Let's do this later when we make it configurable. Note that if we expose a filter callback that takes in an entire row, there is no need to specify the `filterableProperties` anymore. The client can easily make it so by appropriate choice of filter.